src/main/java/com/exasol/adapter/dialects/snowflake/SnowflakeSqlDialect.java

```diff
@@ -64,86 +66,69 @@ public Capabilities getCapabilities() {
      */
+
     @Override
     public StructureElementSupport supportsJdbcCatalogs() {
-        return StructureElementSupport.SINGLE;
+        return SINGLE;
     }
 
-    /**
-     * Get the type of support Athena has for schemas.
-     * <p>
-     * Athena knows schemas as a mirror of databases.
-     *
-     * @return always {@link com.exasol.adapter.dialects.SqlDialect.StructureElementSupport#MULTIPLE}
-     *
-     * @see <a href= "https://docs.aws.amazon.com/athena/latest/ug/show-databases.html">SHOW DATABASES documentation</a>
-     */
     @Override
     public StructureElementSupport supportsJdbcSchemas() {
-        return StructureElementSupport.MULTIPLE;
+        return MULTIPLE;
     }
 
     @Override
     public boolean requiresCatalogQualifiedTableNames(final SqlGenerationContext context) {
         return false;
     }
 
-    @Override
-    public boolean requiresSchemaQualifiedTableNames(final SqlGenerationContext context) {
-        return true;
-    }
-
     @Override
     public String applyQuote(final String identifier) {
-        if (identifier.startsWith("_")) {
-            return quoteWithBackticks(identifier);
-        } else {
-            return quoteWithDoubleQuotes(identifier);
-        }
+        return "\"" + identifier + "\"";
     }
 
-    private String quoteWithBackticks(final String identifier) {
-        final StringBuilder builder = new StringBuilder("`");
-        builder.append(identifier);
-        builder.append("`");
-        return builder.toString();
-    }
-
-    private String quoteWithDoubleQuotes(final String identifier) {
-        final StringBuilder builder = new StringBuilder("\"");
-        builder.append(identifier);
-        builder.append("\"");
-        return builder.toString();
+    @Override
+    public boolean requiresSchemaQualifiedTableNames(final SqlGenerationContext context) {
+        return false;
     }
 
     @Override
     public NullSorting getDefaultNullSorting() {
-        return NullSorting.NULLS_SORTED_AT_END;
+        return NullSorting.NULLS_SORTED_AT_START;
     }
 
     @Override
     protected RemoteMetadataReader createRemoteMetadataReader() {
         try {
-            return new AthenaMetadataReader(this.connectionFactory.getConnection(), this.properties);
+            return new SnowflakeMetadataReader(this.connectionFactory.getConnection(), this.properties);
         } catch (final SQLException exception) {
-            throw new RemoteMetadataReaderException("Unable to create Athena remote metadata reader.", exception);
+            throw new RemoteMetadataReaderException("Unable to create Snowflake remote metadata reader.", exception);
         }
     }
 
     @Override
     protected QueryRewriter createQueryRewriter() {
-        return new BaseQueryRewriter(this, createRemoteMetadataReader(), this.connectionFactory);
+        return new BaseQueryRewriter(this, this.createRemoteMetadataReader(), this.connectionFactory);
     }
+
+    @Override
```

Comment:
This looks suspiciously like the [default handling](https://github.com/exasol/virtual-schema-common-jdbc/blob/84d0b720b154c393a939099ba3fcb8177166942a/src/main/java/com/exasol/adapter/dialects/AbstractSqlDialect.java#L146), with the exception that it does not handle `NULL`. Please check it that is necessary. Remove if not.